### PR TITLE
helm: Properly support passing subnet-tags/subnet-ids/instance-tags filters as a list

### LINKED ISSUE #21179: v1.12.0 subnetIDsFilter (subnet-ids-filter) cannot be parsed properly on Helm Chart values.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

My Helm Chart values:

```yaml
eni:
  enabled: true
  subnetIDsFilter: "subnet-1234a, subnet-1234b, subnet-1234c"
  updateEC2AdapterLimitViaAPI: true 
  awsReleaseExcessIPs: true 
```

Then I found out only `subnet-1234c` would be parsed in the logs of `cilium-operator` (see below).

```
level=info msg="  --subnet-ids-filter='subnet-1234a,,subnet-1234b,,subnet-1234c'" subsys=cilium-operator-aws

(...ignore)

level=warning msg="Instance not found! Please delete corresponding ciliumnode if instance has already been deleted." instanceID=i-123xfff name=ip-10-10-205-106.ap-northeast-1.compute.internal subsys=ipam
level=warning msg="Instance not found! Please delete corresponding ciliumnode if instance has already been deleted." instanceID=i-123xxx name=ip-10-10-217-40.ap-northeast-1.compute.internal subsys=ipam
(...etc)
```

Yes, have you noticed? There are double commas `,` in the end of every subnet IDs, so I tried to following config -- (removed the commas and replaced to spaces.

```yaml
subnetIDsFilter: "subnet-1234a subnet-1234b subnet-1234c"
```

And it worked as expected. 

I think this is a bug that the `cilium-operator` mis-parserd the subnet IDs, please help to verify it, thanks.

### Cilium Version

v1.12.0

### Kernel Version

Linux ip-10-218-232-237.ap-northeast-1.compute.internal 5.4.129-62.227.amzn2.x86_64 #1 SMP Wed Jul 7 00:08:43 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

v1.20 AWS EKS

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by slayer321**
Signed-off-by: slayer321 <sachin.maurya7666@gmail.com>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Fixes: #21179

Used `join` in sprig to pass the list a string.

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-configmap.yaml [hunk 1]
```diff
@@ -405,9 +405,15 @@ data:
 {{- end }}
   ec2-api-endpoint: {{ .Values.eni.ec2APIEndpoint | quote }}
   eni-tags: {{ .Values.eni.eniTags | toRawJson | quote }}
-  subnet-ids-filter: {{ .Values.eni.subnetIDsFilter | quote }}
-  subnet-tags-filter: {{ .Values.eni.subnetTagsFilter | quote }}
-  instance-tags-filter: {{ .Values.eni.instanceTagsFilter | quote }}
+{{- if .Values.eni.subnetIDsFilter }}
+  subnet-ids-filter: {{ .Values.eni.subnetIDsFilter | join "," | quote }}
```
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml]:
```suggestion
  subnet-ids-filter: {{ .Values.eni.subnetIDsFilter | join " " | quote }}
```

File: install/kubernetes/cilium/templates/cilium-configmap.yaml [hunk 2]
```diff
@@ -405,9 +405,15 @@ data:
...
+{{- end }}
+{{- if .Values.eni.subnetTagsFilter }}
+  subnet-tags-filter: {{ .Values.eni.subnetTagsFilter |  join "," | quote }}
```
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml]:
```suggestion
  subnet-tags-filter: {{ .Values.eni.subnetTagsFilter |  join " " | quote }}
```

File: install/kubernetes/cilium/templates/cilium-configmap.yaml [hunk 3]
```diff
@@ -405,9 +405,15 @@ data:
...
+{{- end }}
+{{- if .Values.eni.instanceTagsFilter }}
+  instance-tags-filter: {{ .Values.eni.instanceTagsFilter | join "," | quote }}
```
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml]:
```suggestion
  instance-tags-filter: {{ .Values.eni.instanceTagsFilter | join " " | quote }}
```

**Review by gandro — CHANGES_REQUESTED** [review]
Thanks for the PR, looks good overall. The values need to be space separated however.

Also, could you update the title to something a bit more descriptive for the release notes, e.g. "helm: Properly support passing subnet-tags/subnet-ids/instance-tags filters as a list"

**Comment by gandro** [comment]
Great, thanks! One last request: Please squash all commits into one. Thank you.

**Review by tommyp1ckles — APPROVED** [review]

**Review by gandro — APPROVED** [review]

**Review by gandro — CHANGES_REQUESTED** [review]
@slayer321 Seems like there are some inconsistencies, the Helm smoketest failed


```
 error: cilium/values.yaml has been modified
Make sure to apply your changes to cilium/values.yaml.tmpl and run 'make -C install/kubernetes cilium/values.yaml' to regenerate.
```

**Comment by slayer321** [comment]
What is can see is there is no changes needs to be in the [cilium/values.yaml.tmpl](https://github.com/cilium/cilium/blob/master/install/kubernetes/cilium/values.yaml.tmpl) file. Am I  missing something here ?

**Comment by gandro** [comment]
> What is can see is there is no changes needs to be in the [cilium/values.yaml.tmpl](https://github.com/cilium/cilium/blob/master/install/kubernetes/cilium/values.yaml.tmpl) file. Am I missing something here ?

The `values.yaml` file is generated from the `values.yaml.tmpl` file. You will have to edit the latter fist, and then regenerate the values.yaml file using `make -C install/kubernetes cilium/values.yaml`

These are the lines you want to change:
https://github.com/cilium/cilium/blob/14c66c7bc5c6012eb6289dc0726f59dd2f16c137/install/kubernetes/cilium/values.yaml.tmpl#L593-L605

**Comment by gandro** [comment]
Change looks good, please squash your commits again. Thanks!

**Review by gandro — APPROVED** [review]

**Comment by gandro** [comment]
@slayer321 Could you squash again, please?

**Comment by gandro** [comment]
Helm and docs smoketest passed. No need to run full CI on those flags, since they are not covered. Marking ready to merge.

**Pull request merged**